futaba/cogs/moderation/core.py

```diff
@@ -46,8 +46,8 @@ def setup(self):
         pass
 
     @staticmethod
-    def build_reason(ctx, action, minutes, reason, past=False):
-        full_reason = StringBuilder(f"{action} by {user_discrim(ctx.author)}")
+    def build_reason(message, action, minutes, reason, past=False):
```

Comment:
I would do what you did for `perform_jail` with this as well as other commands use this as well and use `ctx`.
That or change them to use message.